signer-cloud-server/src/main/java/com/wultra/signercloud/server/signer/SignerService.java

```diff
@@ -107,4 +115,50 @@ private Signer.SignerBuilder updateSigner(final Signer signer) {
         return signer.toBuilder()
                 .timestampLastUpdated(Instant.now());
     }
+
+    /**
+     * Updates the {@link SignerStatus} of a {@link Signer}.
+     *
+     * @param externalSignerId identifier of the signer to update
+     * @param request request containing the new status
+     * @return result of operation as {@link Try}
+     */
+    @Transactional
+    Try updateStatus(final String externalSignerId, final UpdateSignerStatusRequest request) {
+        try {
+            updateStatus(externalSignerId, request.signerStatus());
+            return Try.success();
+        } catch (SignerNotFoundException | SignerStatusTransitionException | RestClientException e) {
+            return Try.error(e);
+        }
+    }
+
+    private void updateStatus(final String externalSignerId, final SignerStatus newStatus) throws RestClientException {
+        var signer = signerRepository.findByExternalSignerId(externalSignerId)
+                .orElseThrow(() -> new SignerNotFoundException("Signer not found for external signer ID: " + externalSignerId));
+
+        final var oldStatus = signer.getStatus();
+
+        if (oldStatus == newStatus) {
+            return;
+        }
+
+        final var isTransitionValid = VALID_STATUS_TRANSITIONS.getOrDefault(oldStatus, Collections.emptySet())
+                .contains(newStatus);
+
+        if (!isTransitionValid) {
+            throw new SignerStatusTransitionException("Invalid status transition from " + oldStatus + " to " + newStatus);
```

Comment:
You may use the formatter.
```suggestion
            throw new SignerStatusTransitionException("Invalid status transition from %s to %s".formatted(oldStatus, newStatus);
```

Reply:
Updated